datafusion/physical-plan/src/joins/hash_join/partitioned_hash_eval.rs

```diff
@@ -87,12 +132,15 @@ impl Hash for HashExpr {
     fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
         self.on_columns.dyn_hash(state);
         self.description.hash(state);
+        self.seeds().hash(state);
     }
 }
 
 impl PartialEq for HashExpr {
     fn eq(&self, other: &Self) -> bool {
-        self.on_columns == other.on_columns && self.description == other.description
+        self.on_columns == other.on_columns
+            && self.description == other.description
+            && self.seeds() == other.seeds()
```

Comment:
I think there was also a bug lurking here where expressions would erroneously compare equal even if they had different `RandomState`s. I don't think it was likely to result in a real bug but let's fix it while were here.